src/Development/IDE/Types/Location.hs

```diff
@@ -10,151 +10,63 @@ module Development.IDE.Types.Location
     , readSrcSpan
     ) where
 
 import Control.Applicative
 import Language.Haskell.LSP.Types (Location(..), Range(..), Position(..))
-import Control.DeepSeq
 import Control.Monad
-import Data.Binary
-import Data.Maybe as Maybe
-import Data.Hashable
+import Data.Binary (Binary)
+import Control.DeepSeq (NFData)
+import Data.Hashable (Hashable, hash)
 import Data.String
-import qualified Data.Text as T
 import FastString
-import Network.URI
-import System.FilePath
-import qualified System.FilePath.Posix as FPP
-import qualified System.FilePath.Windows as FPW
-import System.Info.Extra
 import qualified Language.Haskell.LSP.Types as LSP
-import Language.Haskell.LSP.Types as LSP (
-    filePathToUri
-  , NormalizedUri(..)
-  , Uri(..)
-  , toNormalizedUri
-  , fromNormalizedUri
-  )
 import SrcLoc as GHC
 import Text.ParserCombinators.ReadP as ReadP
-import GHC.Generics
 
-
--- | Newtype wrapper around FilePath that always has normalized slashes.
--- The NormalizedUri and hash of the FilePath are cached to avoided
--- repeated normalisation when we need to compute them (which is a lot).
---
--- This is one of the most performance critical parts of ghcide, do not
--- modify it without profiling.
-data NormalizedFilePath = NormalizedFilePath NormalizedUriWrapper !Int !FilePath
-    deriving (Generic, Eq, Ord)
-
-instance NFData NormalizedFilePath where
-instance Binary NormalizedFilePath where
-  put (NormalizedFilePath _ _ fp) = put fp
-  get = do
-    v <- Data.Binary.get :: Get FilePath
-    return (toNormalizedFilePath v)
-
-
-instance Show NormalizedFilePath where
-  show (NormalizedFilePath _ _ fp) = "NormalizedFilePath " ++ show fp
-
-instance Hashable NormalizedFilePath where
-  hash (NormalizedFilePath _ h _) = h
-
--- Just to define NFData and Binary
-newtype NormalizedUriWrapper =
-  NormalizedUriWrapper { unwrapNormalizedFilePath :: NormalizedUri }
-  deriving (Show, Generic, Eq, Ord)
-
-instance NFData NormalizedUriWrapper where
-  rnf = rwhnf
-
-
-instance Hashable NormalizedUriWrapper where
+newtype NormalizedFilePath = NormalizedFilePath LSP.NormalizedFilePath
+  deriving (NFData, Binary, Show, Hashable, Eq)
 
 instance IsString NormalizedFilePath where
     fromString = toNormalizedFilePath
 
 toNormalizedFilePath :: FilePath -> NormalizedFilePath
 -- We want to keep empty paths instead of normalising them to "."
-toNormalizedFilePath "" = NormalizedFilePath (NormalizedUriWrapper emptyPathUri) (hash ("" :: String)) ""
-toNormalizedFilePath fp =
-  let nfp = normalise fp
-  in NormalizedFilePath (NormalizedUriWrapper $ filePathToUriInternal' nfp) (hash nfp) nfp
+toNormalizedFilePath "" = NormalizedFilePath $ LSP.NormalizedFilePath emptyPathUri ""
+toNormalizedFilePath fp = NormalizedFilePath $ LSP.toNormalizedFilePath fp
 
 fromNormalizedFilePath :: NormalizedFilePath -> FilePath
-fromNormalizedFilePath (NormalizedFilePath _ _ fp) = fp
+fromNormalizedFilePath (NormalizedFilePath nfp) =  LSP.fromNormalizedFilePath nfp
```

Comment:
I would prefer a version with `'`. That is consistent with `uriToFilePath'` and it has the advantage that it’s easy to grep for all uses of the unsuffixed version to audit whether they are fine. Currently I need to take imports into account which makes grep pretty useless.